### PR TITLE
[YUNIKORN-2797] Increase handlers.go test coverage

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
Increase test coverage and some refactor.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2797

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/webservice/handlers_test.go [hunk 1]
```diff
@@ -318,6 +318,18 @@ func newApplication(appID, partitionName, queueName, rmID string, ugi security.U
...
+	req, err := http.NewRequest("GET", "", strings.NewReader(baseConf))
+	assert.NilError(t, err, "Error creating request")
+	resp := &MockResponseWriter{}
+	getStackInfo(resp, req)
+
+	// assert stack trace
+	assert.Assert(t, strings.Contains(string(resp.outputBytes), "goroutine"), "Stack trace should be present in the response")
+	assert.Assert(t, strings.Contains(string(resp.outputBytes), "test"), "Stack trace should be present in the response")
+	assert.Assert(t, strings.Contains(string(resp.outputBytes), "github.com/apache/yunikorn-core/pkg/webservice.getStackInfo"), "Stack trace should be present in the response")
+}
```
Comment (manirajv06) [inline pkg/webservice/handlers_test.go:331]: Can you cover -ve cases?

File: pkg/webservice/handlers_test.go [hunk 2]
```diff
@@ -1358,17 +1364,42 @@ func TestGetPartitionNodes(t *testing.T) {
...
+	getPartitionNode(resp, req)
+	assertParamsMissing(t, resp)
+
+	// Test partition not exist
+	req, err = createRequest(t, "/ws/v1/partition/notexists/node/node-1", map[string]string{"partition": "notexists"})
+	assert.NilError(t, err, "Get Nodes for PartitionNode Handler request failed")
+	resp = &MockResponseWriter{}
+	getPartitionNodes(resp, req)
+	assertPartitionNotExists(t, resp)
+}
```
Comment (manirajv06) [inline pkg/webservice/handlers_test.go:1396]: It is growing big. Probably, time to split into two separate methods, one for Nodes and another one for specific node.

**Review by manirajv06 — CHANGES_REQUESTED** [review]

File: pkg/webservice/handlers_test.go [hunk 3]
```diff
@@ -1358,17 +1364,42 @@ func TestGetPartitionNodes(t *testing.T) {
...
 	assertNodeIDNotExists(t, resp)
+
+	// Test param missing
+	req, err = http.NewRequest("GET", "/ws/v1/partition/default/node", strings.NewReader(""))
+	assert.NilError(t, err, "Get Node for PartitionNode Handler request failed")
+	resp = &MockResponseWriter{}
+	getPartitionNode(resp, req)
+	assertParamsMissing(t, resp)
+
+	// Test partition not exist
```
Comment (pbacsko) [inline pkg/webservice/handlers_test.go:1390]: nit: "does not exist"

File: pkg/webservice/handlers_test.go (hunk 2 above)
Comment (ryankert01) [reply pkg/webservice/handlers_test.go:1396]: Done.

File: pkg/webservice/handlers_test.go (hunk 3 above)
Comment (ryankert01) [reply pkg/webservice/handlers_test.go:1390]: Done.

File: pkg/webservice/handlers_test.go (hunk 1 above)
Comment (ryankert01) [reply pkg/webservice/handlers_test.go:331]: Is there a negative case for getStackInfo? It seems to be cannot make runtime.Stack() failed. 

Ps. I added a deep stack trace case.

Comment (manirajv06) [reply pkg/webservice/handlers_test.go:350]: I meant, asserting cases like StatusInternalServerError etc by creating such situation.

Comment (ryankert01) [reply pkg/webservice/handlers_test.go:352]: @manirajv06 Just to clarify, I'm actually asking how to create a negative case (such as StatusInternalServerError). The whole API doesn't need any parameter, and the function mainly use `runtime.Stacktrace`, not sure if there's a way to fail it...

Comment (manirajv06) [reply pkg/webservice/handlers_test.go:352]: StatusInternalServerError is returned only when go built in write() method fails. Yes, not required to cover the case. Just to be safer, we can assert status code against valid response code (200).

Comment (ryankert01) [reply pkg/webservice/handlers_test.go:354]: Got it, done.

**Review by manirajv06 — APPROVED** [review]
looks good. Will merge after successful run.

**Comment by ryankert01** [comment]
I got it! We shouldn't assert statusOK in this function's test because it will not set statusOK in the slope.

**Comment by manirajv06** [comment]
> I got it! We shouldn't assert statusOK in this function's test because the tested function will not set statusOK in the slope.

It should other way around, not equal to StatusInternalServerError. Anyways, not a problem.

**Pull request closed without merging**